src/ApolloProvider.tsx

```diff
@@ -43,6 +43,14 @@ export default class ApolloProvider extends Component<ProviderProps, any> {
   constructor(props, context) {
     super(props, context);
 
+    this._init(props);
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
I think this should be `componentDidUpdate`. We also wouldn’t need to pass in `props` in that case.